src/middlewared/middlewared/plugins/disk.py

```diff
@@ -207,32 +207,35 @@ def decrypt(self, job, devices, passphrase=None):
     async def __get_smartctl_args(self, devname):
         camcontrol = await camcontrol_list()
         if devname not in camcontrol:
-            raise CallError(f"Camcontrol did not report device {devname}")
+            return
 
         args = await get_smartctl_args(devname, camcontrol[devname])
         if args is None:
-            raise CallError(f"Unable to get smartctl args for device {devname}")
+            return
 
         return args
 
     @private
     async def toggle_smart_off(self, devname):
         args = await self.__get_smartctl_args(devname)
-        await run('/usr/local/sbin/smartctl', '--smart=off', *args, check=False)
+        if args:
+            await run('/usr/local/sbin/smartctl', '--smart=off', *args, check=False)
 
     @private
     async def toggle_smart_on(self, devname):
         args = await self.__get_smartctl_args(devname)
-        await run('/usr/local/sbin/smartctl', '--smart=on', *args, check=False)
+        if args:
+            await run('/usr/local/sbin/smartctl', '--smart=on', *args, check=False)
 
     @private
     async def serial_from_device(self, name):
         args = await self.__get_smartctl_args(name)
-        p1 = await Popen(['smartctl', '-i'] + args, stdout=subprocess.PIPE)
-        output = (await p1.communicate())[0].decode()
-        search = re.search(r'Serial Number:\s+(?P<serial>.+)', output, re.I)
-        if search:
-            return search.group('serial')
+        if args:
+            p1 = await Popen(['smartctl', '-i'] + args, stdout=subprocess.PIPE)
+            output = (await p1.communicate())[0].decode()
+            search = re.search(r'Serial Number:\s+(?P<serial>.+)', output, re.I)
+            if search:
+                return search.group('serial')
```

Comment:
If smart cant be detected I would think we have to try and get the serial using `geom`.

virtio-block can have serials which in this case would be empty.

Reply:
@william-gr why not always use `geom` in the first place then?

Reply:
@themylogin because geom cannot get the serial from "especial" devices (e.g. mfid, rocket raid, etc)